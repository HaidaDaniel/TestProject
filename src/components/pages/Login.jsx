/** @format */
import { Container, Row, Col } from 'react-bootstrap'

import { Header, LoginBody } from '../index'

const Login = () => (
  <Container>
    <Row>
      <Col>
        <Header></Header>
      </Col>
    </Row>
    <Row>
      <Col>
        <LoginBody></LoginBody>
      </Col>
    </Row>
  </Container>
)

export default Login

Login.propTypes = {}
