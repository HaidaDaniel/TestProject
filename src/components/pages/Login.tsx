/** @format */
import { Container, Row, Col } from 'react-bootstrap'

import { Header, LoginBody } from '../'

const Login = () => (
  <Container>
    <Row>
      <Col>
        <Header />
      </Col>
    </Row>
    <Row>
      <Col>
        <LoginBody />
      </Col>
    </Row>
  </Container>
)

export default Login
