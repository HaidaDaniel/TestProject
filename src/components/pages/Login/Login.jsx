/** @format */

import LoginBody from '../../LoginBody/LoginBody'

function Login() {
  return (
    <>
      <LoginBody></LoginBody>
    </>
  )
}
export default Login
