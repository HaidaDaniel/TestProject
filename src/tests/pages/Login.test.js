import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
<<<<<<< HEAD
<<<<<<<< HEAD:src/tests/components/pages/Login.test.js

import Login from '../../../components/pages/Login';

import wrapComponentInAllProviders from '../../helpers/wrapComponentInAllProviders';


describe('Login Component', () => {
    it('should render login page', () => {
========
=======
>>>>>>> cdcb47e (1st refactor of tests description)
import wrapComponentInAllProviders from '../../tests/helpers/wrapComponentInAllProviders';
import Login from '../../pages/Login';

describe('Login Component', () => {
    it('should renders login page', () => {
<<<<<<< HEAD
>>>>>>>> origin/newTest:src/tests/pages/Login.test.js
=======
>>>>>>> cdcb47e (1st refactor of tests description)
        render(wrapComponentInAllProviders(<Login />));
        expect(document.body).toMatchSnapshot();

        const headerElement = screen.getByText('Shop');
        expect(headerElement).toBeInTheDocument();
        const loginElement = screen.getByTestId('login col');
        expect(loginElement).toBeInTheDocument();

    });
});
