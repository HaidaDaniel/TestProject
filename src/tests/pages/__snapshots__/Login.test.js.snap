// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<<< HEAD:src/tests/__snapshots__/components/pages/Login.snap.js
exports[`Login Component should render login page 1`] = `
========
exports[`Login Component should renders login page 1`] = `
>>>>>>>> origin/newTest:src/tests/pages/__snapshots__/Login.test.js.snap
<body>
  <div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <nav
            class="sc-jrkPcm hELBzT navbar navbar-expand-lg navbar-light"
          >
            <a
              class="sc-kDnyCx ctmCGs mx-4"
              href="/"
            >
              Shop
            </a>
            <button
              aria-controls="basic-navbar-nav"
              aria-label="Toggle navigation"
              class="navbar-toggler collapsed"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="sc-ipMvLY dWioSl justify-content-end mx-2 navbar-collapse collapse"
              id="basic-navbar-nav"
            >
              <div
                class="ml-auto mx-2 navbar-nav"
              >
                <a
                  class="nav-link"
                  href="/registration"
                >
                  Registration
                </a>
                <a
                  class="nav-link"
                  href="/login"
                >
                  Login
                </a>
              </div>
            </div>
          </nav>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="row"
          >
            <div
              class="sc-iJfdHH dECjJl"
            >
              <div
                class="sc-gisvNi khLjsh col-md-6 col-12"
                data-testid="login col"
              >
                <h3
                  class="sc-ezGUZh HKjiX"
                >
                  Login
                </h3>
                <form
                  action="#"
                >
                  <div
                    class="sc-bYUneI chbYQP"
                  >
                    <label
                      class="sc-kMbQoj bqKgrV form-label"
                    >
                      Email address
                    </label>
                    <input
                      class="sc-ilpitK IVHGj form-control"
                      name="email"
                      placeholder="Email address"
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    class="sc-bYUneI chbYQP"
                  >
                    <label
                      class="sc-kMbQoj bqKgrV form-label"
                    >
                      Password
                    </label>
                    <input
                      class="sc-ilpitK IVHGj form-control"
                      name="password"
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                  </div>
                  <button
                    class="mb-3 btn btn-primary"
                    data-testid="submit-button"
                    type="submit"
                  >
                    Login
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
